### ISSUE TITLE
Minus figures are read the same as Positive figures in Excel and PDFs

### ISSUE THREAD

**Issue by jaselson** [nvaccess/nvda#12366]
### Steps to reproduce:
I have an excel table containing minus figures (entered as accounting with the £ sign on the left). NVDA does not read them as minus figures. Thus £40,000 and -£40,000 are read as the same. NVDA does not read the minus. I have exported the excel sheet as an accessible PDF to go on our website. NVDA has the same problem with minus figures in the PDF. I work for a Local Government Authority and we are required by law to provide any PDFs in an accessible PDF. Currently we would be breaking the law and could be fined if I publish a PDF with minus figures which are not being read correctly by any user using NVDA. The law has been in affect since last September and will affect any local Authority (Parish, District, Borough, Unitary or County Council in the UK).
### Actual behavior:
Minus figures are read the same as positive figures in Excel and PDFs created from Excel
### Expected behavior:
Minus figures to be read correctly in Excel and Adobe Acrobat
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2020.4
#### Windows version:
Windows 10 Pro 64-bit
#### Name and version of other software in use when reproducing the issue:
Excel 2016 and Adobe Acrobat Pro DC 2021.001.20150
#### Other information about your system:
Dell XPS
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If add-ons are disabled, is your problem still occurring?
N/A
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Not tried

**Comment by feerrenrut** [comment]
I can't reproduce this issue with Excel for Office 365. After resetting my NVDA config to default settings, I tried:
- entering "-£40,000" in cell A1 it is read as "minus forty thousand A1"
- Setting the cell formatting to currency, same result.
- Setting the cell formatting to text, same result.

Some things to check:
- Change your symbol level to ensure you are not set to None, use keyboard shortcut: `NVDA+p`?
- Which synthesizer are you using? I tested with both espeak and Windows onecore

If none of these things help, please supply an example excel spreadsheet. You can add this to a zip file and attach that to a comment. This will help by ensuring we are at least looking at the same data.

**Comment by feerrenrut** [comment]
I have been able to reproduce this issue. Only if I set the "Negative Numbers:" option in Excel, cell formatting, currency to the second option which has no minus symbol but the text in red.

![image](https://user-images.githubusercontent.com/3491346/117235654-b3fda900-ae59-11eb-8dd5-bd79839cd2cd.png)

All other options (that have a minus symbol displayed visually) result in the minus symbol being spoken. For this option, the user would have to use NVDA+F to report the formatting, and hear that the text is red.

Really, excel should be providing the minus symbol to screen readers (such as NVDA). This issue also affects Narrator.

While we certainly agree this is important, it is unlikely to be fixed soon. We might be able to workaround the issue in NVDA by checking the number formatting, and then checking the color of the text and adding a minus symbol, but this is error prone.

**To make this document accessible now, change the negative number format to one that includes a minus symbol.**

I strongly suspect doing this, then exporting to PDF will resolve the issue. However, this may not be the formatting you wanted.

**Comment by jaselson** [comment]
Thanks both, you have solved my problem. I had set the cell formatting to Accounting which places "-£" to the left of the cell with the number on the right as I visually preferred it. This doesn't give any options for negative numbers but does display them with a minus. I then set conditional formatting to colour any negative numbers differently. 

Setting the cell formatting to "Number" and choosing the formatting for negative numbers works correctly. 

It's a shame setting the cell format to Accounting doesn't work though as I've been persuading my customer that all the cells in my table need figures for accessibility (i.e. £0 rather than blank cells, and £- doesn't read out) and to not put negative numbers in parenthesis as they would prefer with Accounts. This has been my biggest challenge - making the standard format for a balance sheet accessible as my customer says adding £ signs and £0 amounts makes the information too busy.

Is there any best practice for accounts and balance sheets you could point me to?

**Comment by jaselson** [comment]
Sorry, I said both people when it was just the one person (who was super helpful :))

**Comment by feerrenrut** [comment]
Ah, I see. Yes, I can also reproduce the issue with the accounting category. Sorry, I don't have any best practices for you, though you may want to look at the "currency" cell formatting option.

I'm going to make some edits to the initial description, so that we can track this bug.

**Comment by feerrenrut** [comment]
@jaselson When using the "accounting" option, it is possible for the screen reader to announce the "-$" symbols. Setting symbol mode to "all" with `NVDA + p` will read them as "dash dollar". The minus symbol is interpreted as a dash because it is not next to a number. 

I still think this is a bug, it is putting visual presentation before accuracy in an application where accuracy really counts.

**Comment by britechguy** [comment]
This seems closely related to this feature request from 2018:  https://github.com/nvaccess/nvda/issues/8875 

That feature request was subsequent to this topic on the Groups.io NVDA Group:  https://nvda.groups.io/g/nvda/topic/27393769

It is actually shocking to me that this issue remains unresolved almost 3 years later.  The use of standard visual formats is common practice in accounting and those formats tell the person reading them by sight information about whether the value itself is positive or negative.  In a screen reader, if the value of the cell (whether numeric or currency) is negative, it must be announced as such if accuracy for the user is to be maintained.  It doesn't matter how it's presented visually without the actual negative sign, the value itself is negative, and that's what matters here.

**Comment by britechguy** [comment]
> This needs to be reported to Microsoft Office.

Has this been done?  It bears a lot more weight coming from NVAccess as an organization, and direct reports to MS have been done in the past.

**Comment by feerrenrut** [comment]
We'll do what we can, in the mean-time we still recommend giving feedback directly via feedback hub.

**Comment by britechguy** [comment]
@Adriani90 

The problem being that Accounting format does not include minus signs, period.  In accounting, a negative value is indicated by enclosure in parentheses.  This has been accounting convention since before I was born.

If the value in the cell is -125.37, and formatting is Accounting, then what's displayed is $(125.37).  That is never, ever going to change, as it is accepted accounting convention.

Any screen reader that's dealing with spreadsheets needs to account for negative values for something like this.  There is no negative sign shown at all in accounting format.

Currency format is a different issue, and you can choose to have a minus sign in front of the dollar sign or use something that looks exactly like Accounting format where parentheses indicate a negative value, and this is whether or not red color is applied as well.  You can also have just red applied for a negative value.

What is critical is the cell value, which is what a screen reader should communicate, rather than its presentation format.  If I'm working with some spreadsheet where negative numbers are in use, I must know that the value I'm on is, indeed, negative, regardless of the formatting applied.  The value is far more important, in context, than the format used to present it.

And both Accounting and Currency formats apply only to numeric values.  When I apply Accounting format in Excel 2016 there is no option for a minus sign, only a specification of decimal places, and negatives are always indicated visually via enclosure in parentheses.  If I apply Currency format, and select the option that employs a minus sign, there is no space between the minus sign and the leading dollar sign.  All the remaining currency options use either a red color, enclosure in parentheses after the dollar sign, or both in combination to denote a negative number.

I have never seen anything otherwise, and unless Microsoft changed the default formats in later versions of Excel, which would be highly unusual, it should still be this way.

**Comment by XLTechie** [comment]
CC @michaelDCurran

**Comment by Adriani90** [comment]
While this is true for the english language, it is not always the case, depending in which language your windows is. You can even change how negative figures are displayed in Windows settings / clock / numbers. MS Excel uses the Windows setting natively to display negative values when choosing accounting cell format.

However, I've discovered that enabling UIA for Excel in NVDA settings reports the negative number with minus even when accounting format is chosen.

**Comment by britechguy** [comment]
> However, I've discovered that enabling UIA for Excel in NVDA settings reports the negative number with minus even when accounting format is chosen.

That's great, and language issues aside, if UIA for Excel is not the default NVDA setting, this is still a problem.  However it does it for any given language community, if a number is negative that needs to be clearly indicated when working with Excel, even if the format itself does not use the minus symbol as the method.  That was really my central point and concern.  How a numeric value is formatted is important as far as how it's presented to the sighted, but when you're actually working with those numbers directly it's their value that's critical.  Knowing that something is negative/less than zero/a debit value rather than positive/greater than zero/a credit value is just critical in the milieu of Excel.

**Comment by britechguy** [comment]
I am bumping this again, mostly to see if there is any "behind the scenes" status or additional information available.

If I format any currency cell using any of the formats other than the one with the leading minus sign, none are read as negative or debit.  This has been a problem for years now, and the visually the use of parentheses to enclose a number is the negative/debit indicator.  It's been around longer than I've been alive.

All of the negative dollar figures  presented this way read as though they were plain old positive equivalents.  Even the parentheses that can be seen (literally) are not read.

Is NVDA not being given any information whatsoever about the actual cell value other than what's on display?  If that's the case, then why are values enclosed in parentheses not having something announced that indicates their presence.

It's impossible to know whether this is an NVDA issue, an Excel issue, or a combination of the two without knowing what gets handed to NVDA to announce when one lands in one of these cells.  I find it odd that something styled as "-$1234.10" gets read (using OneCore) as "minus one thousand two hundred thirty-four dollars and 10 cents," while all other formats, whether the negative is indicated by enclosing in parentheses, the use of red color, or the use of both at the same time all get treated as positive and get read as, "one thousand two hundred thirty-four dollars and 10 cents."   I have no idea if NVDA has access to the actual numeric value separate from how it's presented visually, but if such access is available that needs to hold sway over the announcement of the value to the user.

JAWS, by contrast, announces the value with the leading negative sign correctly (as does NVDA), the one that's presented only in read incorrectly as though it's a positive number, but both of the formats that use enclosure by parentheses have "Left paren," the dollar value, "Right paren" announced, which does allow someone who's familiar with accounting formatting to know they're dealing with a debit value (negative value) because of this.

Narrator is "the best of the bunch" in that it announces the number with a leading minus sign plus both the red or black parentheses enclosed values all as negative.

No one is 100% accurate in that the format that uses only the red color to denote negative value is always announced as positive by NVDA, JAWS, and Narrator.
